README.md

```diff
@@ -1161,6 +1161,19 @@ Note that options in configuration file are just the same options aka switches u
 
 You can use `--ignore-config` if you want to disable all configuration files for a particular yt-dlp run. If `--ignore-config` is found inside any configuration file, no further configuration will be loaded. For example, having the option in the portable configuration file prevents loading of home, user, and system configurations. Additionally, (for backward compatibility) if `--ignore-config` is found inside the system configuration file, the user configuration is not loaded.
 
+### Specifying encoding of config files
+
+By default, config files are read by UTF-8.
+If you saved your config file in a different encoding than that, you may write any of these to the first line of the file:
+
+1. `# -*- coding: ENCODING -*-` (e.g. `# -*- coding: gbk -*-`)
+1. `# coding: ENCODING` (e.g. `# coding: shift-jis`)
+1. `# vi: set fileencoding=ENCODING` (e.g. `# vi: set fileencoding=euc-kr`)
+
+There must not be any characters before that, including spaces.
+
+To use UTF-32 and UTF-16, save your file with BOM enabled. The method above cannot be used in this case.
```

Comment:
Why can't prefix be used for this?

Reply:
Because they encode even ASCII characters differently so you wouldn’t be able to parse the `# coding` line properly without some charset detector.

Reply:
ASCII chars only get a null character in the front. We could easily add that to our regex as I understand

Reply:
UTF-16 and UTF-32 intercepts byte sequence differently than others. for example, if we have `0x23 0x20` (`b'# '`), it will be `⌠` instead of `# `. on UTF-32 in this example, it's an error as the length of input isn't multiple of 4